core/roboconf-messaging-rabbitmq/src/main/java/net/roboconf/messaging/rabbitmq/RabbitMqClientAgent.java

```diff
@@ -383,6 +385,40 @@ public void listenToTheDm( ListenerCommand command ) throws IOException {
 	}
 
 
+	@Override
+	public String getMessagingType() {
+		return MessagingConstants.FACTORY_RABBIT_MQ;
+	}
+
+
+	@Override
+	public Map<String, String> getConfiguration() {
+		final Map<String, String> configuration = new LinkedHashMap<>();
+		configuration.put(MESSAGING_TYPE_PROPERTY, MessagingConstants.FACTORY_RABBIT_MQ);
+		configuration.put(MessagingConstants.RABBITMQ_SERVER_IP, this.messageServerIp);
+		configuration.put(MessagingConstants.RABBITMQ_SERVER_USERNAME, this.messageServerUsername);
+		configuration.put(MessagingConstants.RABBITMQ_SERVER_PASSWORD, this.messageServerPassword);
+		return Collections.unmodifiableMap(configuration);
+	}
+
+
+	@Override
+	public boolean setConfiguration( final Map<String, String> configuration ) {
+		final boolean result;
+		final String type = configuration.get(MESSAGING_TYPE_PROPERTY);
+		final String ip = configuration.get(MessagingConstants.RABBITMQ_SERVER_IP);
+		final String username = configuration.get(MessagingConstants.RABBITMQ_SERVER_USERNAME);
+		final String password = configuration.get(MessagingConstants.RABBITMQ_SERVER_PASSWORD);
+		if (MessagingConstants.FACTORY_RABBIT_MQ.equals(type) && ip != null && username != null && password != null) {
```

Comment:
Username and password can be null.
In that case, it means we will use the default values "guest/guest".

Same thing for ip => localhost.